### PR TITLE
fix: Issue 402 (GroovyBracketInsertionCompleter shall not treat a ${ as a closure))

### LINKED ISSUE #402: GroovyBracketInsertionCompleter shall not  treat a `${` as a closure
## Situation
A `${` does lead to the situation that the bracket will be closed this way:
```
${
   CURSOR
} 
```

## Wanted
Typing A `${`  shall lead to  
```
${CURSOR}
```

## Solution
Change the beahviour of `GroovyBracketInsertionCompleter` accordingly.

Remark: This was mentioned at https://github.com/de-jcup/eclipse-jenkins-editor/issues/53 as well - because JenkinsEditor uses EGradle groovy editor under the hood. So implementing this will fix the issue as well.

### PR DESCRIPTION
**Pull request by BeckerFrank**
This Pull request 
- closes #402

### CODE REVIEW

**Review by de-jcup — CHANGES_REQUESTED** [review]
Hello @BeckerFrank ,
thank you for contributing again!

Your solution works well, but I have some minor change requests. See my comments.
After changes are done I will merge your PR.

PS: I changed your PR description a bit  - so the PR 
- is automatically linked with the issue
- issue texts are shown completely. 
- merging the PR will automatically close the issue

See my text changes there (it's just a `This PR closes #$nr`)

File: egradle-plugin-main/src/main/java-eclipse/de/jcup/egradle/eclipse/ui/GroovyBracketInsertionCompleter.java [hunk 1]
```diff
@@ -82,7 +82,20 @@ public void keyReleased(KeyEvent e) {
             EclipseUtil.logError("Cannot get contentType", ex);
             return;
         }
-        EclipseUtil.safeAsyncExec(new Runnable() {
+		try {
+			String charBeforBracket = document.get(offset - 2, 1);
```
[the remark this answers is not in the document]
Comment (BeckerFrank) [reply egradle-plugin-main/src/main/java-eclipse/de/jcup/egradle/eclipse/ui/GroovyBracketInsertionCompleter.java:86]: Done!
See next commit!

File: egradle-plugin-main/src/main/java-eclipse/de/jcup/egradle/eclipse/ui/GroovyBracketInsertionCompleter.java [hunk 2]
```diff
@@ -82,7 +82,20 @@ public void keyReleased(KeyEvent e) {
...
+			if (charBeforBracket.equals("$") && e.character == '{') {
+				document.replace(offset , 0, "}");
+				selectionProvider.setSelection(new TextSelection(offset, 0));
+				return;
+			}
+		} catch (BadLocationException e2) {
+			e2.printStackTrace();
```
[the remark this answers is not in the document]
Comment (BeckerFrank) [reply egradle-plugin-main/src/main/java-eclipse/de/jcup/egradle/eclipse/ui/GroovyBracketInsertionCompleter.java:93]: Done!
See next commit!

**Review by de-jcup — APPROVED** [review]
Looks fine for me.

**Comment by de-jcup** [comment]
@BeckerFrank : I will squash and merge by one commit - so I can shorten the messages (they are longer than 50 characters ) see https://git-scm.com/docs/git-commit#_discussion for details. But you will still be the author of the change inside the commit history.

I do all of my commits in such way:

```
This is my short message (up to 50 chars max)

- here are detailed informations (the empty line before makes formatting in command line much better)
- you calso use some simple markup here. At least github does render this very good and it's good readable
- eclipse does give you also hints about this behaviour inside the `commit message` field of the `git staging` view
```

**Pull request merged**